ui/src/app/views/concept-table/component.tsx

```diff
@@ -135,10 +143,35 @@ export class ConceptTable extends React.Component<{concepts: Concept[];
 
   rowExpansionTemplate(data) {
     return (<SynonymsObject>
-      {fp.uniq(data.conceptSynonyms).join(', ')}
+      {this.highlightWithSearchTerm(fp.uniq(data.conceptSynonyms).join(', '))}
     </SynonymsObject>);
   }
 
+  highlightWithSearchTerm(stringToHighlight: string) {
+    const {searchTerm} = this.props;
+    if (searchTerm) {
+      const words: string[] = [];
+      let searchWords = searchTerm.split(new RegExp(',| '));
+      searchWords = searchWords.filter(w => w.length > 0 );
+      searchWords = searchWords.map(word => word.replace(/[&!^\/\\#,+()$~%.'":*?<>{}]/g, ''));
+      const matchString = new RegExp(searchWords.join('|'), 'i');
+      const matches = stringToHighlight.match(new RegExp(matchString, 'gi'));
+      const splits = stringToHighlight.split(new RegExp(matchString, 'gi'));
+      if (matches) {
+        for (let i = 0; i < matches.length; i++) {
+          words.push(splits[i], matches[i]);
+        }
+        words.push(splits[splits.length - 1]);
+      }
+      return words.map(word => <span
+        style={matchString.test(word.toLowerCase()) ? styles.highlighted : {}}>
+          {word}
+        </span>);
+    } else {
+      return stringToHighlight;
```

Comment:
nit: we can also write the above as 
 if (!searchTerm) {
return stringToHighlight;
}
const words....